akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -283,13 +364,19 @@ import scala.util.control.NonFatal
         // The only invariant we need to keep is to never emit more requests than the downstream emitted so far.
         @tailrec def bufferDemand(n: Long): Unit = {
           val current = get()
-          if (current eq PassThrough) real.request(n)
-          else if (!compareAndSet(current, Buffering(current.demand + n))) bufferDemand(n)
+          if (current eq PassThrough) {
+            if (VirtualProcessor.Debug) println(s"VirtualPublisher#${VirtualProcessor.this.hashCode}WrappedSubscription($real).bufferDemand($n) passthrough")
+            real.request(n)
+          } else if (!compareAndSet(current, Buffering(current.demand + n))) {
+            if (VirtualProcessor.Debug) println(s"VirtualPublisher#${VirtualProcessor.this.hashCode}WrappedSubscription($real).bufferDemand($n) buffering")
+            bufferDemand(n)
+          }
```

Comment:
this is only adding debug logging

Reply:
Oh, sorry, didn't catch this, you think I should remove it?

Reply:
no, was only an observation for future reviewers